File: src/main/java/com/rtomyj/skc/enums/ErrorType.java
package com.rtomyj.skc.enums;


public enum ErrorType
{
	G001("URL or data in body doesn't use proper syntax")
	, D001("Requested resource was not found")
	, D002("Error occurred interfacing with resource(s)");

	private final String error;

	ErrorType(String error) { this.error = error; }

	@Override
	public String toString() { return this.error; }
}

File: src/main/java/com/rtomyj/skc/exception/ErrorType.kt
package com.rtomyj.skc.exception

import lombok.ToString
import org.springframework.http.HttpStatus

@ToString
enum class ErrorType(val error: String, val httpStatus: HttpStatus) {
	G001("URL or data in body doesn't use proper syntax", HttpStatus.NOT_FOUND)

	, D001("Requested resource was not found", HttpStatus.NOT_FOUND)
	, D002("Error occurred interfacing with resource(s)", HttpStatus.INTERNAL_SERVER_ERROR);
}